src/odemis/gui/log.py

```diff
@@ -202,3 +205,26 @@ def write_to_field(self):
                 self.textfield.Remove(0, first_new)
 
         self.textfield.Refresh()
+
+state_change_function_list = [] #List for passing component.name to the function stage_change_pop_up
+def observe_comp_state (comps_with_role):
+    '''
+    Function which produces an warning/information pop-up in the OS if an error or recovery of an error occurs
+    :param comps_with_role: list with all the components and their data
+    '''
+    for component in comps_with_role:
+        def state_change_pop_up(component_state_value, component_name=component.name):
+            if component_state_value == ST_RUNNING:
+                show_message(None, 'Recovered ' + component_name,
+                             'Functionality of the "' + component_name + '" is recovered successfully',
+                             timeout=3.0, level=logging.INFO)
+
+            elif isinstance(component_state_value, HwError):
+                show_message(None, 'Error in ' + component_name, str(component_state_value),
+                             timeout=5.0, level=logging.WARNING)
+
+        state_change_function_list.append(state_change_pop_up)
```

Comment:
Add a comment explaining why you keep a reference to them -> so that they are not dereferenced (as VAs use weakrefs).